### ISSUE TITLE
Control the tag/theme color via CMS

### ISSUE THREAD

**Issue by mae1884** [mae1884/nipper_ny#5]
Tags should use the theme color as set in the CMS. Perhaps this color should not be set by the CMS in terms of best practice. Still, the tags should be in CAPS, using sans-serif font, and should appear within the article in the Gatsby build.

**Comment by marcomiduri** [comment]
@mae1884 Can you explain about this?

**Comment by mae1884** [comment]
@marcomiduri  similar to the last one: see the tags on the homepage 
![image](https://user-images.githubusercontent.com/61389274/114954060-56a8a600-9e1f-11eb-90c3-16dcf2d5c167.png)

The tags are not styled like on albanycountypost.com

![image](https://user-images.githubusercontent.com/61389274/114954105-6aeca300-9e1f-11eb-8bd5-e9bc864d8b5f.png)

Also, the author image/avatar is displaying oddly.  
![image](https://user-images.githubusercontent.com/61389274/114954173-94a5ca00-9e1f-11eb-86b1-b77a1e8f62a6.png)

"Theme color" refers to this https://albanycountypost.com/ghost/#/settings?showBrandingModal=true

![image](https://user-images.githubusercontent.com/61389274/114954264-c74fc280-9e1f-11eb-8659-933d4eb60276.png)


If this should rather be set in the code, just tell me.

**Comment by marcomiduri** [comment]
@mae1884 Do you only need tag text color to be set with this accent color? thanks for providing these screenshots. I will update the avatar style.

**Comment by mae1884** [comment]
> 
> 
> @mae1884 Do you only need tag text color to be set with this accent color? thanks for providing these screenshots. I will update the avatar style.

We have not decided on the details of what color, what font, etc. Just wondering if it is possible to control it via the Ghost CMS, no matter the color. If not, we will have to wait.

**Comment by marcomiduri** [comment]
@mae1884 Yes, we can apply all these settings(Branding) on our frontend. Please let me know all Ghost settings you want to implement on our frontend.

**Comment by mae1884** [comment]
> 
> 
> @mae1884 Yes, we can apply all these settings(Branding) on our frontend. Please let me know all Ghost settings you want to implement on our frontend.

I know you can apply the settings, but are you saying they can be set via the Ghost CMS? if we change the color 5 weeks from now, will it update properly in the Gatsby build? or do I have to tell you the specific color settings and you code it in?

**Comment by marcomiduri** [comment]
@mae1884 You can only set it on GhostCMS Settings and trigger rebuild. Then the new color will be set. I just finished this. Sorry, I forgot to push the updates to the new branch. Updates are pushed to the master branch and it's on live website now. Please check it. I also updated the avatar image issue. Thanks

**Comment by mae1884** [comment]
> 
> 
> @mae1884 You can only set it on GhostCMS Settings and trigger rebuild. Then the new color will be set. I just finished this. Sorry, I forgot to push the updates to the new branch. Updates are pushed to the master branch and it's on live website now. Please check it. I also updated the avatar image issue. Thanks

Yes that is what I was trying to do. Wow. I just changed the color and it rebuilt on nipperny.com. You're great. Thank you. Can you point me to where in the code you connect the CMS to the front-end like that?

**Comment by marcomiduri** [comment]
Great, thank you. We can get all GhostCMS settings data using `allGhostSettings` query.

https://github.com/mae1884/nipper_ny/blob/88cb438606e267ab291ad46575d04c37625dac2c/src/templates/index.js#L65-L69

**Comment by mae1884** [comment]
Ah, so it's Ghost API -- GraphQL -- Gatsby, @marcomiduri do we need to be concerned with how many queries we run in terms of performance? 

Related: when the site is "rebuilt" does it rebuild every piece of content? When we migrate content from our old site, we will migrate 20,000+ posts, each with photos. Is there any consideration we need to make for how to handle that? We are a news site.

> 
> 
> Great, thank you. We can get all GhostCMS settings data using `allGhostSettings` query.
> 
> https://github.com/mae1884/nipper_ny/blob/88cb438606e267ab291ad46575d04c37625dac2c/src/templates/index.js#L65-L69

**Comment by marcomiduri** [comment]
@mae1884 Queries will be run when the website built so it doesn't affect actual web pages. As you mentioned, if there are a lot of blogs, it will take some time to build the whole site. Netlify doesn't build only the updated things.

**Comment by mae1884** [comment]
> 
> 
> @mae1884 Queries will be run when the website built so it doesn't affect actual web pages. As you mentioned, if there are a lot of blogs, it will take some time to build the whole site. Netlify doesn't build only the updated things.

@marcomiduri Could that be a problem if we have 20, 30, 40,000+ posts?

**Comment by mae1884** [comment]
> 
> 
> > @mae1884 Queries will be run when the website built so it doesn't affect actual web pages. As you mentioned, if there are a lot of blogs, it will take some time to build the whole site. Netlify doesn't build only the updated things.
> 
> @marcomiduri Could that be a problem if we have 20, 30, 40,000+ posts?

We need to be able to publish to the live site relatively quickly. 15 minutes to build is ok. Longer is not ok.

**Comment by marcomiduri** [comment]
@mae1884 I don't think so. As our project does SSR, it takes some time to build the pages on the server-side. I can't promise how many minutes it will take... It depends on the packages, dependencies, and project size

**Comment by mae1884** [comment]
> 
> 
> @mae1884 I don't think so. As our project does SSR, it takes some time to build the pages on the server-side. I can't promise how many minutes it will take... It depends on the packages, dependencies, and project size

Are we able to do this

> Netlify doesn't build only the updated things.

Can we do this?
https://www.netlify.com/blog/2020/04/23/enable-gatsby-incremental-builds-on-netlify/?utm_source=blog&utm_medium=gatsby-perf-jl&utm_campaign=devex

+     "build": "cross-env GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES=true gatsby build --log-pages"

**Comment by marcomiduri** [comment]
@mae1884 I have used this on Gatsby v3 mostly and it worked fine but there were some issues on v2. Our project is using v2 so I don't recommend this unless we use v3.

**Comment by marcomiduri** [comment]
@mae1884 If you agree to upgrade the package versions to the latest, I will do that and set incremental build for you.
